slither/tools/read_storage/read_storage.py

```diff
@@ -208,6 +244,58 @@ def get_target_variables(self, **kwargs) -> None:
             if slot_info:
                 self._slot_info[f"{contract.name}.{var.name}"] = slot_info
 
+    def find_constant_slot_storage_type(
+        self, var: StateVariable
+    ) -> Tuple[Optional[str], Optional[int]]:
+        """
+        Given a constant bytes32 StateVariable, tries to determine which variable type is stored there, using the
+        heuristic that if a function reads from the slot and returns a value, it probably stores that type of value.
+        Also uses the StorageSlot library as a heuristic when a function has no return but uses the library's getters.
+        Args:
+            var (StateVariable): The constant bytes32 storage slot.
+
+        Returns:
+            type (str): The type of value stored in the slot.
+            size (int): The type's size in bits.
+        """
+        if not (var.is_constant and str(var.type) == "bytes32"):
+            return None
+        storage_type = None
+        size = None
+        funcs = []
+        for c in self.contracts:
+            funcs.extend(c.get_functions_reading_from_variable(var))
+        fallback = [f for f in var.contract.functions if f.is_fallback]
+        funcs += fallback
+        for func in funcs:
+            if func.return_type is not None:
+                rets = func.return_type
+                for ret in rets:
+                    size, _ = ret.storage_size
+                    if size <= 32:
+                        return str(ret), size * 8
+            for node in func.all_nodes():
+                exp = node.expression
+                # Look for use of the common OpenZeppelin StorageSlot library
+                if f"getAddressSlot({var.name})" in str(exp):
+                    return "address", 160
+                if f"getBooleanSlot({var.name})" in str(exp):
+                    return "bool", 1
+                if f"getBytes32Slot({var.name})" in str(exp):
+                    return "bytes32", 256
+                if f"getUint256Slot({var.name})" in str(exp):
+                    return "uint256", 256
+                # Look for variable assignment in assembly loaded from a hardcoded slot
+                if (
+                    isinstance(exp, AssignmentOperation)
+                    and isinstance(exp.expression_left, Identifier)
+                    and isinstance(exp.expression_right, CallExpression)
+                    and "sload" in str(exp.expression_right.called)
+                    and exp.expression_right.arguments[0] == var.expression
+                ):
+                    return "address", 160
+        return storage_type, size
```

Comment:
I'm not really sure how this will hold up to any contract on mainnet. I'd almost rather leave the type as bytes32... What is this used for?